### ISSUE TITLE
Avoid using braces in interpolation false positive

### ISSUE THREAD

**Issue by jacehensley-wf** [dart-lang/linter#234]
The `Avoid using braces in interpolation when not needed` reports a lint when it shouldn't. Specifically when the variable starts with a `$`.

``` dart
var $someString = 'Some Value';
var concatenated = 'Value is: ${$someString}'; // Currently lints when it shouldn't.
```

```dart
var $someString = 'Some Value';
var concatenated = 'Value is: $$someString'; // Analyzer Warning
```

**Comment by alexeieleusis** [comment]
https://github.com/dart-lang/linter/pull/235

**Comment by pq** [comment]
Thanks!

Fixed w/ https://github.com/dart-lang/linter/commit/0f6c9caa76d744fc1c4e1544c01ba7819b62a9c2
